x/crosschain/keeper/cctx.go

```diff
@@ -93,6 +93,10 @@ func (k Keeper) setNonceToCCTX(
 	}
 }
 
+const (
+	TSIndexKey = "cctx-ts-"
+)
+
 // SetCrossChainTx set a specific cctx in the store from its index
 func (k Keeper) SetCrossChainTx(ctx sdk.Context, cctx types.CrossChainTx) {
```

Comment:
We should want to set it at cctx creation? Should not be in this method but in place that create the cctx for the first time

Reply:
Right now we use `SetCrossChainTx()` for both create and update. I could separate the codepaths but that would increase the size of this PR quite a bit.

But it might be worth it to avoid updating the index ever iteration.

Reply:
I agree separating create and update paths would be a decent chunk of refactor, I don't know if it's worth it for just this value.
The cctx is created using the `NewCCTX` function, and is but only saved to the state after validation. An alternative could be to add the counter to cctx object itself.
I think all the options have pros and cons and we should decide on the one that is the simplest to test ,